support/notifications-client/client.go

```diff
@@ -68,38 +85,42 @@ type Notification struct {
 	Modified    int          `json:"modified,omitempty"` // The last modification timestamp
 }
 
-// Send a POST call to the notifications service
-func (nc NotificationsClient) RecieveNotification(n Notification) error {
+// Send a notification to the notifications service
+func (nc NotificationsClient) SendNotification(n Notification) error {
 	client := &http.Client{}
 
 	// Get the JSON request body
 	requestBody, err := json.Marshal(n)
 	if err != nil {
-		fmt.Println(err.Error())
 		return err
 	}
 
 	// Create the request
-	req, err := http.NewRequest(http.MethodPost, nc.RemoteUrl, bytes.NewBuffer(requestBody))
-	req.Header.Add("Content-Type", "application/json")
+	remoteNotificationServiceUrl := fmt.Sprintf(UrlPattern, nc.NotificationServiceHost, nc.NotificationServicePort, NotificationApiPath)
+
+	return doPost(remoteNotificationServiceUrl, bytes.NewBuffer(requestBody), client)
+}
+
+// Function to do post request
+func doPost(url string, binaryReqBody io.Reader, client *http.Client) error {
+	req, err := http.NewRequest(HttpPostMethod, url, binaryReqBody)
+	req.Header.Add(ContentType, ContentTypeJsonVal)
+
 	if err != nil {
-		fmt.Println(err.Error())
 		return err
 	}
 
-	// Asynchronous call
-	go makeRequest(client, req)
```

Comment:
You are changing the call to be made asynchronous. This will change the performance of the function on the caller, to complete mostly in no time to make a complete http request in place.
This should not be approved without the approval from some of the core microservices developers.

Reply:
@feclare It's not about performance, the real thing is : this code style in my test could not trigger a real http request to server. You can run my test with http mock server to see the result. This file shows the author is `Ryan Comer, Dell`, who can contact with he? The code style can not get http error, just return a `nil`. This function has no test, has wrong method name. @drasko can you arbitrate this or run my test case?

Reply:
No sense in spawning a goroutine if sucess/failure and/or response can not be communicated back to the calling code. I would go with the synchronous call here.

I do not think that `makeRequest` is a wrong name, however (I have seen this pattern Go code already - example InfluxDB's Telegraf: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/httpjson/httpjson.go#L64).

Reply:
It is possible that author wanted to achieve this: https://medium.com/dev-bits/making-concurrent-http-requests-in-go-programming-language-823b51bb1dc2 - but without communicating result via channel, current call does not make much sense.

Reply:
@drasko 

* to your first comment:
 Actually, I want to say the original `ReceiveNotification` is a wrong name (not `makeRequest`). I just want to indicate the exists and original code is not completely right.

* to your second comment:
I totally agree with you, actually the old code style did not use a channel to wait the http request finish and can not get the response.